### PR TITLE
Disable GPP stub when `initialize=false` is set until `Fides.init()` is called

### PR DESCRIPTION
**Pull request by gilluminate**
Closes [PROD-2214](https://ethyca.atlassian.net/browse/PROD-2214)

### Description Of Changes

Initializing GPP without initializing Fides has been known to cause issues, especially in cases where Fides never gets initialized.

* When FidesJS is downloaded with `initialize=false`, ensure that GPP is not initialized at all
* When `Fides.init()` is called, ensure that the GPP stub is initialized promptly and is not delayed by any other blocking logic (e.g. API calls, etc.)

### Code Changes

* eliminated waste on resulting `fides.js` response by not printing config object twice.
* include `gppEnabled` as part of the config options object (similar to existing `tcfEnabled`)
* add new `FidesInitializing` event for benefit of GPP initialization and other similar use cases
* wait for `FidesInitializing` to be dispatched before initializing GPP.
* Add and clean up related Cypress tests

### Steps to Confirm

1. Open browser console
2. visit demo page with GPP enabled but `initialize` set to false (eg. `/fides-js-demo.html?gpp=true&initialize=false`)
3. observe browser console message that Fides was not initialized
4. check if GPP is initialized by attempting to ping it `__gpp('ping', (data) => {console.log(data)})`
5. observe browser console error that `__gpp` does not exist yet.
6. initialize Fides using the browser console `window.Fides.init()`
7. observe browser console debug messages indicating that Fides started normally
8. check if GPP is initialized by attempting to ping it `__gpp('ping', (data) => {console.log(data)})`
9. observe that the ping now displays a response indicating that GPP has been initialized.

image
image

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [x] documentation complete
* [x] Issue Requirements are Met
* [ ] Update `CHANGELOG.md`

[PROD-2214]: https://ethyca.atlassian.net/browse/PROD-2214?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/privacy-center/pages/api/fides-js.ts [hunk 1]
```diff
@@ -289,18 +288,13 @@ export default async function handler(
     document.head.append(style);
     `
       : ""
-  }${
-    isTestMode // let end-to-end tests set the config and initialize as needed
-      ? ""
-      : `
```
Comment (gilluminate) [inline clients/privacy-center/pages/api/fides-js.ts:295]: we don't need `isTestMode` anymore, now that we have `intialize=false`

File: clients/privacy-center/pages/api/fides-js.ts [hunk 2]
```diff
@@ -289,18 +288,13 @@ export default async function handler(
...
-    var fidesConfig = ${fidesConfigJSON};
```
Comment (gilluminate) [inline clients/privacy-center/pages/api/fides-js.ts:296]: no longer prints the config twice

File: clients/privacy-center/public/fides-js-components-demo.html [hunk 3]
```diff
@@ -10,20 +10,20 @@
...
+        }${gppEnabled ? "&gpp=true" : ""}${
           !!params.size ? `&${params.toString()}` : ""
         }`;
 
         document.write('<script src="' + src + '"><\/script>');
       })();
-      (function () {
-        if (window.fidesConfig?.experience?.gpp_settings?.enabled)
-          document.write('<script src="./lib/fides-ext-gpp.js"><\/script>');
-      })();
```
Comment (gilluminate) [inline clients/privacy-center/public/fides-js-components-demo.html:26]: handled using query params

**Review by NevilleS — APPROVED** [review]
This is great. Thanks!

**Pull request merged**